### PR TITLE
Debug Enrollment errors

### PR DESCRIPTION
**Pull request by directionless**
Replace error.New with error.Wrap

Add stacktraces to exit conditions

### CODE REVIEW

File: pkg/osquery/extension.go [hunk 1]
```diff
@@ -330,14 +330,15 @@ func (e *Extension) Enroll(ctx context.Context) (string, bool, error) {
...
+	// note that we set invalid two ways. Via the return, _or_ via isNodeInvaliderr
 	keyString, invalid, err := e.serviceClient.RequestEnrollment(ctx, e.Opts.EnrollSecret, identifier, enrollDetails)
 	if isNodeInvalidErr(err) {
 		invalid = true
 	} else if err != nil {
 		return "", true, errors.Wrap(err, "transport error in enrollment")
 	}
 	if invalid {
-		return "", true, errors.Wrap(err, "enrollment invalid")
+		return "", true, errors.Errorf("enrollment invalid (orig_err: %v)", err)
```
Comment (blaedj) [inline pkg/osquery/extension.go]: don't we lose the backtrace etc if return a new error here?

Comment (directionless) [reply pkg/osquery/extension.go]: Yeah. I'm thinking I'm going to have to do something really ugly

Comment (directionless) [reply pkg/osquery/extension.go]: Okay. How's this look

Comment (blaedj) [reply pkg/osquery/extension.go]: much better 😃

**Comment by directionless** [comment]
@blaedj is `if Rails.env.production? || ENV.fetch('USE_STACKDRIVER', false)` better?

**Pull request merged**